Add cancel dialog event listener & Export FeedbackService, FeedbackDirective

workfel opened the pull request:
- I need service to force re-render the screenshot, because before set screen image i need to hide all private user information.

workfel committed:
add cancel dialog listener and export service, directive

artar94 commented:
@workfel I think it would be great to make another service specifically for export.
It should have only one method `feedback(): Observable<Feedback>`, which, if successful, returns `Feedback`, and if canceled, returns an error.
This will allow you to use this library with any buttons and directives, not just the built-in one, and it will not force the user to deal with the internal organization of the initScreenshotCanvas, setFeedback, openFeedbackDialog, etc. methods.

workfel commented:
ok, but why don't juste export the current Service ?

dintecom commented:
I have nothing against it, but I would recommend at least renaming the current `FeedbackService` to `FeedbackInternalService` so that the new service has a user-friendly name.

artar94 commented:
I have nothing against it, but I would recommend at least renaming the current `FeedbackService` to `FeedbackInternalService` so that the new service has a user-friendly name.

workfel committed:
send error message when dialog is canceled and rename feedback service

workfel commented:
Ok, i just updated the PR

workfel committed:
solve bug 'The operation is insecure' for external images https://github.com/RickonChen/feedback/issues/6

workfel committed:
add image for CORS

workfel committed:
Update README and package to use new NPM repository

workfel committed:
-u

workfel committed:
add angular LIB

workfel committed:
add reamde on npm

workfel committed:
add missing peerDep package

workfel committed:
add link to repo git on npm package

workfel committed:
fix bug multiple send event when multiple directive is on app

workfel committed:
feat(modal): Can disabled send button if description field is empty

workfel committed:
update readme

workfel committed:
fix(build): Add missing dependencies html2canvas

workfel committed:
feat(description): Can directly set description field

workfel committed:
remove angular/material as dependencies

workfel committed:
fix:  add delay before taking init screenshot

spike008t committed:
fix(dep): fix peer-dependency angular < 9

spike008t committed:
chore: upgrade angular core 7

spike008t committed:
chore: update codelyzer

spike008t committed:
chore: update material 7

spike008t committed:
chore: update angular 8

spike008t committed:
chore: browserslist

spike008t committed:
chore: upgrade angular 8

spike008t committed:
chore: update package-lock.json

spike008t committed:
fix(browser): remove not for IE

spike008t committed:
fix(build)): set target es5 and module es2015

spike008t committed:
chore: add support IE11

spike008t committed:
chore(readme): updated

spike008t committed:
Merge pull request #1 from spike008t/fix/angular-9-compat

WIP: Fix/angular 8 & 9 compat - bump 2.0.0

spike008t committed:
chore: setup nvmrc

spike008t committed:
Merge pull request #4 from spike008t/fix/nvmrc

chore: setup nvmrc

spike008t committed:
chore: add @digikare namespace on package

spike008t committed:
Merge pull request #5 from spike008t/fix/digikare-namespace-package

chore: add @digikare namespace on package

spike008t committed:
chore: update package.json

spike008t committed:
fix(build): update target es2015 to fix issue with string size limit

Looks like on es5 the string has a limit size of ~2Mb..
The image generated is cropped..

spike008t committed:
chore: on feedback, open window with the image

spike008t committed:
chore(release): bump v2.0.1

spike008t committed:
Merge pull request #6 from spike008t/fix/buffer-size-image

Fix/buffer size image & bump v2.0.1

dgk-zartha committed:
chore(upgrade): ng v9

dgk-zartha committed:
chore(upgrade): ng v9

dgk-zartha committed:
chore(upgrade): ng material v9

dgk-zartha committed:
chore(upgrade): bump tslib to 2.3.1

dgk-zartha committed:
chore(upgrade): ng v10

dgk-zartha committed:
chore(upgrade): add browserlist

dgk-zartha committed:
chore(upgrade): material v10

dgk-zartha committed:
chore(upgrade): remove build-ng-packagr

dgk-zartha committed:
chore(upgrade): ng v11

dgk-zartha committed:
chore(upgrade): ng v12

dgk-zartha committed:
chore(upgrade): material v12

dgk-zartha committed:
chore(upgrade): ng v13

dgk-zartha committed:
chore(upgrade): build lib

dgk-zartha committed:
build(bump): v2.2.0

dgk-zartha committed:
build(bump): v2.2.0

workfel committed:
Merge pull request #7 from digikare/upgrade-to-v13

Upgrade to v13
